client/js/jsonedit.js

```diff
@@ -2049,21 +2054,32 @@ const clickButton = async function(event) {
   }
 }
 
+let widgetCoordCache = null;
 window.addEventListener('mousemove', function(e) {
   if(!jeEnabled)
     return;
-  jeState.mouseX = Math.floor((e.clientX - roomRectangle.left) / scale);
-  jeState.mouseY = Math.floor((e.clientY - roomRectangle.top ) / scale);
+  const x = jeState.mouseX = Math.floor((e.clientX - roomRectangle.left) / scale);
+  const y = jeState.mouseY = Math.floor((e.clientY - roomRectangle.top ) / scale);
+
+  if(!jeZoomOut && x > 1600 || jeMouseButtonIsDown)
+    return;
+
+  if(!widgetCoordCache) {
+    widgetCoordCache = [];
+    for(const [ widgetID, widget ] of widgets) {
```

Comment:
Is `widgetID` needed?

Reply:
Guess not. Was copy-pasted from the old version.